src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -482,7 +524,13 @@ public void setNetworkTimeout(Executor executor, int milliseconds) throws SQLExc
     @Override
     public int getNetworkTimeout() throws SQLException {
         checkClosed();
-        throw new ReadReplicaUnsupportedOperationException();
+        if(compatibleWithPreviousVersion) {
+            throw new ReadReplicaUnsupportedOperationException();
+        } else {
+            return connectionProvider
+                .getWriteConnection(new RouteDecisionBuilder(Reason.RW_API_CALL))
```

Comment:
I'm still missing any sanity tests for all of those methods.
* It'll prevent regression this implementation,
* will be the starting point for any other changes related to those methods
* sustain current level of code coverage

Reply:
What cases/behavior would you like to see being tested?